kernel/filesystem/Filesystem.cpp

```diff
@@ -175,6 +176,28 @@ Result filesystem_mkdir(Path *path)
     return result;
 }
 
+Result filesystem_rmdir(Path *path)
+{
+    Directory *directory = directory_open(path_as_string(path), OPEN_READ);
```

Comment:
The Directory object is a higher-level concept, please only use functions from kernel/filesystem to ovoid race condition, cache, and deadlocks. Another option is to implement this function only in userland inside libsystem.